<?xml version="1.0"?>
<launch>
    <!-- log levels and verbosity -->
    <env name="ROSCONSOLE_FORMAT" value="[${time}] [${node}-${severity}] : ${message}" />

    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find snd_description)/urdf/snd_robot.urdf.xacro'"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

    <rosparam command="load" file="$(find snd_control)/config/control.yaml"/>
    
    <node name="controller_spawner" pkg="controller_manager" type="spawner" output="screen"
        args="snd_joint_publisher snd_velocity_controller --shutdown-timeout 1">
    </node>

    <node pkg="snd_control" type="snd_control_ros_node" name="snd_control_node" output="screen">
        <param name="update_rate" value="20.0"/>
    </node>

    <node pkg="topic_tools" type="relay" name="cmd_vel_relay"
        args="cmd_vel snd_velocity_controller/cmd_vel"/>

    <node pkg="topic_tools" type="relay" name="odom_relay"
        args="snd_velocity_controller/odom /odom"/>
</launch>
